drive/snippets/drive_v2/team_drive_snippets/recover_team_drives.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// [START drive_recover_team_drives]
+
+/**
+ * Find all shared team drives without an organizer and add one.
+ * @param{string} userId user ID
+ * @return{obj} page token
+ * */
+async function recoverTeamDrives(userId) {
+  const {GoogleAuth} = require('google-auth-library');
+  const {google} = require('googleapis');
+
+  // Get credentials and build service
+  // TODO (developer) - Use appropriate auth mechanism for your app
+  const auth = new GoogleAuth({scopes: 'https://www.googleapis.com/auth/drive'});
+  const service = google.drive({version: 'v2', auth});
+  const drives = [];
+  const newOrganizerPermission = {
+    type: 'user',
+    role: 'organizer',
+    value: 'user@example.com',
+  };
+  newOrganizerPermission.value = userId;
+
+  let pageToken = null;
+  try {
+    const res = await service.teamdrives.list({
+      q: 'organizerCount = 0',
+      fields: 'nextPageToken, items(id, name)',
+      useDomainAdminAccess: true,
+      pageToken: pageToken,
+    });
+    Array.prototype.push.apply(drives, res.data.items);
+    res.data.items.forEach(function(drive) {
```

Comment:
Not sure we want to encourage this pattern. This will end up executing all the requests in parallel which, depending on how big the results are, can easily exceed an app's quota. And the results of rhte permission insert calls aren't tracked for failure either -- it's just fire and forget which is also not a pattern we want to encourage (though in this case it's not horrible as the drive will just reappear next time it's run.) 

There are some libraries like https://github.com/sindresorhus/p-throttle or https://github.com/sindresorhus/p-queue that help here, though I prefer to avoid additional dependencies when possible.

Suggest the following:
- Implement paging correctly
- Dot this is two phases -- one pass to page over drives and collect orphaned drives into an array, the 2nd to iterate over the array and add the permissions (rather than doing for each page)
- When adding permissions, add them sequentially and block/await on each request. Add a note acknowledging it is potentially slow and can be improved with parallel requests if throttled correctly.